internal/cli/cmd/cluster/docker.go

```diff
@@ -126,15 +129,158 @@ func NewDockerLoginCmd(hidden bool) *cobra.Command {
 			return fnerrors.New("failed to write %q: %w", cfgFile, err)
 		}
 
-		if nscr := response.NSCR; nscr != nil {
-			fmt.Fprintf(stdout, "\nYou are now logged into your Workspace container registry:\n\n  %s/%s", nscr.EndpointAddress, nscr.Repository)
-			fmt.Fprintf(stdout, "\n\nRun your first build with:\n\n  $ nsc build . -n test:v0.0.1 -p")
+		if _, err := exec.LookPath(credHelperBinary); err != nil {
+			if errors.Is(err, exec.ErrNotFound) {
+				fmt.Fprintf(stdout, "\nWarning: nsc binary is not installed in your $PATH. Docker credentials won't work until you add nsc's path to the $PATH.\n")
+			} else {
+				return fnerrors.New("failed to look up nsc in $PATH: %w", err)
+			}
+
+		} else {
+			if nscr := response.NSCR; nscr != nil {
+				fmt.Fprintf(stdout, "\nYou are now logged into your Workspace container registry:\n\n  %s/%s", nscr.EndpointAddress, nscr.Repository)
+				fmt.Fprintf(stdout, "\n\nRun your first build with:\n\n  $ nsc build . --name test:v0.0.1 --push")
+			}
+
+			fmt.Fprintf(stdout, "\n\nVisit our docs for more details on Remote Builds:\n\n  https://cloud.namespace.so/docs/features/builds\n\n")
+		}
+		return nil
+	})
+
+	return cmd
+}
+
+func NewDockerCredHelperStoreCmd(hidden bool) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "store",
+		Short:  "Unimplemented",
+		Args:   cobra.NoArgs,
+		Hidden: hidden,
+	}
+
+	cmd.RunE = fncobra.RunE(func(ctx context.Context, args []string) error {
+		return fnerrors.New("unimplemented")
+	})
+
+	return cmd
+}
+
+func NewDockerCredHelperGetCmd(hidden bool) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "get",
+		Short:  "Get Workspace's container registry credetial",
+		Args:   cobra.NoArgs,
+		Hidden: hidden,
+	}
+
+	cmd.RunE = fncobra.RunE(func(ctx context.Context, args []string) error {
+		done := console.EnterInputMode(ctx)
+		defer done()
+
+		input, err := readStdin()
+		if err != nil {
+			return fnerrors.New("failed to read from stdin: %w", err)
+		}
+		regURL := string(input)
+
+		resp, err := api.GetImageRegistry(ctx, api.Endpoint)
+		if err != nil {
+			return fnerrors.New("failed to get nscloud registries: %w", err)
+		}
+
+		for _, reg := range []*api.ImageRegistry{resp.Registry, resp.NSCR} {
+			if reg != nil && regURL == reg.EndpointAddress {
+				token, err := fnapi.FetchTenantToken(ctx)
+				if err != nil {
+					return fnerrors.New("failed to fetch tenant token: %w", err)
+				}
+
+				c := credHelperGetOutput{
+					ServerURL: reg.EndpointAddress,
+					Username:  dockerUsername,
+					Secret:    token.Raw(),
+				}
+
+				buf, err := json.Marshal(c)
+				if err != nil {
+					return fnerrors.New("failed to marshal output JSON: %w", err)
+				}
+
+				fmt.Println(string(buf))
+				return nil
+			}
+		}
+
+		return fnerrors.New("credentials not found in nscloud")
```

Comment:
Looking at the docker implementation, it seems that docker surfaces the command output to the user, when there are failures. Which is good.

It also has special-casing for some messages. For examples, if we don't handle the specified server, we should output "credentials not found in native keychain".